src/genai/model.py

```diff
@@ -62,24 +62,38 @@ def generate_stream(
         if len(prompts) > 0 and isinstance(prompts[0], PromptPattern):
             prompts = PromptPattern.list_str(prompts)
 
+        params = self._get_params()
+        params.stream = True
+
         try:
             for i in range(0, len(prompts), Metadata.DEFAULT_MAX_PROMPTS):
                 batch = prompts[i : min(i + Metadata.DEFAULT_MAX_PROMPTS, len(prompts))]
 
-                self.params.stream = True
-                response_gen = self.service.generate(self.model, batch, self.params, options=options, streaming=True)
+                response_gen = self.service.generate(self.model, batch, params, options=options, streaming=True)
+
+                for response in response_gen:
+                    if not response:
+                        continue
 
-                for chunk in response_gen:
-                    if "status_code" in chunk:
-                        error = ast.literal_eval(chunk)
```

Comment:
Not sure why one would ever use `ast.literal_eval` for parsing JSON. Switched to `json`.